src/data/finsupp/basic.lean

```diff
@@ -356,6 +356,44 @@ by { ext, simp [finsupp.single_eq_pi_single, finsupp.equiv_fun_on_fintype], }
 
 end single
 
+/-! ### Declarations about `update` -/
+
+section update
+
+variables [decidable_eq α] [has_zero M] (f : α →₀ M) (a : α) (b : M) [decidable (b = 0)] (i : α)
+
+/-- Replace the value of a `α →₀ M` at a given point `a : α` by a given value `b : M`.
+If `b = 0`, this amounts to removing `a` from the `finsupp.support`.
+Otherwise, if `a` was not in the `finsupp.support`, it is added to it.  -/
+def update : α →₀ M :=
+⟨if b = 0 then f.support.erase a else insert a f.support,
+  function.update f a b,
+  λ i, begin
+    simp only [function.update_apply, ne.def],
+    split_ifs with hb ha ha hb;
+    simp [ha, hb]
+  end⟩
+
+@[simp] lemma update_apply : f.update a b i = function.update f a b i := rfl
+@[simp] lemma update_apply_same : f.update a b a = b := function.update_same a _ _
```

Comment:
I'm surprised the simp linter doesn't reject this

Reply:
Can you add that `f.update I (f I) = f`?

Reply:
I think the simp linter does not reject this because these look different enough to it due to `coe_fn` firing.

I've added the `update_self` lemma.